x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/SecurityIndexReaderWrapper.java

```diff
@@ -81,6 +80,11 @@ public DirectoryReader apply(final DirectoryReader reader) {
                 return reader;
             }
 
+            // permissions are defined, so check if the license actually allows DLS/FLS
+            if (licenseState.checkFeature(Feature.SECURITY_DLS_FLS) == false) {
```

Comment:
At this point in the code we have actually established that the current role applies DLS or FLS, only that there is an `IndicesAccessControl` object.
An `IndicesAccessControl` object in the thread context only means that we performed access checks at some point, it doesn't mean DLS is enabled.

For that we need to check  `permissions.getDocumentPermissions().hasDocumentLevelPermissions()`